### PR TITLE
Reimplement find_decompressed_size without using static-only API

### PR DESCRIPTION
**Pull request by mkitti**
Alternative to #62, that implements `ZSTD_findDecompressedSize` as `find_decompressed_size` in Julia using the public (non-static-only) API.

`ZSTD_getFrameContentSize` only obtains the decompressed size of a single frame. `find_decompressed_size` obtains the accumulated size over multiple frames.

`find_decompressed_size` should continue to respond identically to `ZSTD_findDecompressedSize`.

### CODE REVIEW

File: src/libzstd.jl [hunk 1]
```diff
@@ -163,6 +163,56 @@ end
...
+1. Return a code above if `ZSTD_getFrameContentSize` returns the code.
+2. Return `ZSTD_CONTENTSIZE_ERROR` if `ZSTD_findFrameCompressedSize` errors.
+3. Return `ZSTD_CONTENTSIZE_ERROR` if the frame extends beyond `srcSize`
+"""
+function find_decompressed_size(src::Vector{UInt8})
+    GC.@preserve src find_decompressed_size(pointer(src), length(src))
+end
+function find_decompressed_size(src::Ptr, srcSize::Integer)
+    frameOffset = 0
+    decompressedSize = 0
```
Comment (nhz2) [inline src/libzstd.jl:188]:
```suggestion
    frameOffset = Csize_t(0)
    decompressedSize = Culonglong(0)
```

To ensure the `+=`s later on are type stable

File: src/libzstd.jl [hunk 2]
```diff
@@ -163,6 +163,56 @@ end
...
+
+        # Obtain the decompressed frame content size of the next frame, accumulate
+        frameContentSize = LibZstd.ZSTD_getFrameContentSize(frameSrc, remainingSize)
+        if frameContentSize == ZSTD_CONTENTSIZE_UNKNOWN
+            return ZSTD_CONTENTSIZE_UNKNOWN
+        end
+        if frameContentSize == ZSTD_CONTENTSIZE_ERROR
+            return ZSTD_CONTENTSIZE_ERROR
+        end
+        decompressedSize += frameContentSize
```
Comment (nhz2) [inline src/libzstd.jl:202]: Should this return `ZSTD_CONTENTSIZE_ERROR` if this addition overflows?

File: src/libzstd.jl [hunk 3]
```diff
@@ -163,6 +163,56 @@ end
...
+        # Advance the offset forward by the size of the compressed frame
+        frameCompressedSize = LibZstd.ZSTD_findFrameCompressedSize(frameSrc, remainingSize)
+        if iserror(frameCompressedSize) || frameCompressedSize <= 0
+            return ZSTD_CONTENTSIZE_ERROR
+        end
+        frameOffset += frameCompressedSize
+    end
+
+    # frameOffset > srcSize
+    if frameOffset != srcSize
```
Comment (nhz2) [inline src/libzstd.jl:213]: Does `ZSTD_findFrameCompressedSize` already check for this error? Is that why there is no coverage of this?

**Review by nhz2 — CHANGES_REQUESTED** [review]

File: src/libzstd.jl (hunk 3 above)
Comment (nhz2) [reply src/libzstd.jl:213]: Looking at the docs for `ZSTD_findFrameCompressedSize` it should return an error code if the input is invalid. 
But I guess it is fine to double-check.

File: src/libzstd.jl (hunk 2 above)
Comment (nhz2) [reply src/libzstd.jl:202]:
```suggestion
        decompressedSize, overflow = Base.add_with_overflow(decompressedSize, frameContentSize)
        if overflow
            return ZSTD_CONTENTSIZE_ERROR
        end
```

File: src/libzstd.jl (hunk 3 above)
Comment (mkitti) [reply src/libzstd.jl:216]: I have not figured out how to create a test for this condition yet. My guess is that we may have to create some malformed input.

**Review by nhz2 — APPROVED** [review]

**Comment by mkitti** [comment]
Is this a breaking change?

**Comment by nhz2** [comment]
No, but we can add some downstream tests to check.

**Pull request merged**